### DISCUSSION TITLE
How to speed ​​up a single hhvm shell for some functional tests?

### DISCUSSION THREAD

**Discussion by eternalsakura** [facebook/hhvm#8939] [Q&A] [answered]
Hello, I want to test a lot of hack functions through `./hhvm test.hack`.

I don't need hhvm server, I just want to test from the shell.

But the startup of hhvm is very slow, it takes about one second to test one.
Because I have a lot of hack files to test, this speed is unacceptable :(

How can I speed up the startup of a single hhvm shell?
Or how can I get a hhvm with only terminal shell and no other functions?

**Comment by fredemmott** [comment, marked as answer]
To speed it up, you need a server. It can have a 'CLI server' (communicating with the process you spawn from your shell over a unix socket rather than HTTP requests), but it's still needed.
